goodai/ltm/memory.py

```diff
@@ -282,3 +282,70 @@ def retrieve_from_keywords(self, keywords: list[str]) -> list[RetrievedMemory]:
                 metadata=chunk.metadata,
             ))
         return memories
+
+
+    def temporal_retrieval(self, origin_timestamp: float, temporal_hops: int) -> list[RetrievedMemory]:
+
+        mem: DefaultTextMemory = self.semantic_memory
+        search_index = 0
+        # Get the chunk we are searching from
+        for idx, chunk in enumerate(mem.get_all_chunks()):
+            if chunk.timestamp == origin_timestamp:
+                search_index = idx
+                break
+
+        temporal_chunks = self._timestamp_search(search_index, temporal_hops)
```

Comment:
I would expect the actual search by timestamp to be inside `_timestamp_search`, instead of taking an index.